### PR TITLE
Documents: add sagemaker model building pipeline readthedocs

### PR DESCRIPTION
**Pull request by jerrypeng7773**
*Issue #, if available:*

*Description of changes:*

A RST doc includes usage details about the SageMaker Model Building pipeline.

*Testing done:*

tox -e sphinx

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by ahsan-z-khan — APPROVED** [review]

File: doc/amazon_sagemaker_model_building_pipeline.rst [hunk 1]
```diff
@@ -0,0 +1,872 @@
...
+            on="/",
+            values=[
+                "s3://my-bucket",
+                # from ListTrainingJobsForHyperParameterTuningJob
+                step_tune.properties.TrainingJobSummaries[1].TrainingJobName,
+                "output/model.tar.gz",
+            ],
+    )
+
+The above is just an example, :code:`TuningStep` offers a helper function to generate any :code:`top-k` model data URI easily.
```
Comment (danabens) [inline doc/amazon_sagemaker_model_building_pipeline.rst:264]: will :code:`TuningStep` link to https://sagemaker.readthedocs.io/en/stable/workflows/pipelines/sagemaker.workflow.pipelines.html#sagemaker.workflow.steps.TuningStep ? ideally each of these steps should link to class' documentation

File: doc/amazon_sagemaker_model_building_pipeline.rst [hunk 2]
```diff
@@ -0,0 +1,872 @@
...
+.. _Glue ETL as part of a SageMaker pipeline using Pipeline Callback Step: https://github.com/aws/amazon-sagemaker-examples/blob/main/sagemaker-pipelines/tabular/custom_callback_pipelines_step/sagemaker-pipelines-callback-step.ipynb
+.. _SageMaker Pipelines Lambda Step: https://github.com/aws/amazon-sagemaker-examples/blob/main/sagemaker-pipelines/tabular/lambda-step/sagemaker-pipelines-lambda-step.ipynb
+.. _SageMaker Pipelines integration with Model Monitor and Clarify: https://github.com/aws/amazon-sagemaker-examples/tree/main/sagemaker-pipelines/tabular/model-monitor-clarify-pipelines
+.. _SageMaker Pipelines Tuning Step: https://github.com/aws/amazon-sagemaker-examples/blob/main/sagemaker-pipelines/tabular/tuning-step/sagemaker-pipelines-tuning-step.ipynb
+
+============
+Limitations
+============
+
+The SageMaker Model Building Pipeline python SDK offers abstractions to help you construct a pipeline definition at ease. However, there are certain limitations. The followings are the known limitations, if you found out more, please feel free to open a git issue.
```
Comment (danabens) [inline doc/amazon_sagemaker_model_building_pipeline.rst:822]: `git`->`github issue` (with added link to repo issue creation)

File: doc/amazon_sagemaker_model_building_pipeline.rst [hunk 3]
```diff
@@ -0,0 +1,872 @@
...
+
+.. code-block:: python
+
+    instance_count = ParameterInteger(name="InstanceCount", default_value=2)
+    instance_type = ParameterString(name="InstanceType", default_value="ml.m5.xlarge")
+    process_s3_input_url = ParameterString(name="ProcessingInputUrl")
+
+    processor = Processor(
+       instance_type=instance_count,
+       instance_count=instance_type,
```
Comment (danabens) [inline doc/amazon_sagemaker_model_building_pipeline.rst:841]:
```
instance_type=instance_type,
instance_count=instance_count,
```

File: doc/amazon_sagemaker_model_building_pipeline.rst [hunk 4]
```diff
@@ -0,0 +1,872 @@
...
+        "Step"]
+    )
+
+In summary, pipeline parameter can only be evaluated at runtime, if its actual value needs to be known at compile time in order to either conduct business logic, or apply some operations against, it will lead to an exception.
+
+====================================
+Pipeline Composition Alternative
+====================================
+
+Python SDK provides us with a great abstraction for pipeline composition. Under the hood, it produces a pipeline definition JSON. If you want to author the pipeline definition by hand, you might follow this `JSON Schema`_
```
Comment (danabens) [inline doc/amazon_sagemaker_model_building_pipeline.rst:870]: trailing underscore?

Comment (jerrypeng7773) [reply doc/amazon_sagemaker_model_building_pipeline.rst:870]: this is the link reference, you do `X`_, and then in another line, : .X: https://google.com

File: doc/amazon_sagemaker_model_building_pipeline.rst (hunk 2 above)
Comment (jerrypeng7773) [reply doc/amazon_sagemaker_model_building_pipeline.rst:822]: good point.

File: doc/amazon_sagemaker_model_building_pipeline.rst (hunk 1 above)
Comment (jerrypeng7773) [reply doc/amazon_sagemaker_model_building_pipeline.rst]: using :class: for most cases now.

**Review by navaj0 — APPROVED** [review]

**Review by qidewenwhen — APPROVED** [review]

**Review by danabens — APPROVED** [review]

**Review by shreyapandit — APPROVED** [review]

**Pull request merged**